### PR TITLE
ci: Test with secure boot enabled under mkosi

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

File: man/loader.conf.xml [hunk 1]
```diff
@@ -227,12 +227,33 @@
...
+
+          <varlistentry>
+            <term><option>manual</option></term>
+            <listitem><para>Boot entries for found secure boot keys are created that allow manual
+            enrollment.</para></listitem>
+          </varlistentry>
+
+          <varlistentry>
+            <term><option>if-safe</option></term>
+            <listitem><para>Same behavior with <option>manual</option>, but will try to automatically
```
Comment (daandemeyer) [inline man/loader.conf.xml]: "Same behavior as"

File: man/loader.conf.xml [hunk 2]
```diff
@@ -227,12 +227,33 @@
...
+            enroll the key <literal>auto</literal> if it is considered to be safe. Currently, this is only
+            the case if the system is running inside a virtual machine.</para></listitem>
+          </varlistentry>
+
+          <varlistentry>
+            <term><option>force</option></term>
+            <listitem><para>Always enroll the <literal>auto</literal> key if found. Note that a warning
+            message with a timeout will still be shown if this opertion is unknown to be safe.</para></listitem>
```
Comment (daandemeyer) [inline man/loader.conf.xml]: "operation"

File: src/boot/efi/secure-boot.c [hunk 3]
```diff
@@ -36,19 +36,24 @@ SecureBootMode secure_boot_mode(void) {
 static const char sbat[] _used_ _section_(".sbat") = SBAT_SECTION_TEXT;
 #endif
 
-EFI_STATUS secure_boot_enroll_at(EFI_FILE *root_dir, const char16_t *path) {
+EFI_STATUS secure_boot_enroll_at(EFI_FILE *root_dir, const char16_t *path, bool only_if_safe) {
```
Comment (daandemeyer) [inline src/boot/efi/secure-boot.c]: Maybe invert the boolean? Name it `force` instead of `only_if_safe`?

File: src/boot/efi/secure-boot.h [hunk 4]
```diff
@@ -9,13 +9,14 @@
 typedef enum {
         ENROLL_OFF,         /* no Secure Boot key enrollment whatsoever, even manual entries are not generated */
         ENROLL_MANUAL,      /* Secure Boot key enrollment is strictly manual: manual entries are generated and need to be selected by the user */
+        ENROLL_IF_SAVE,     /* Automatically enroll if it is safe (if we are running inside a VM, for example). */
```
Comment (daandemeyer) [inline src/boot/efi/secure-boot.h]: `ENROLL_IF_SAFE` ?

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

**Comment by medhefgo** [comment]
Apparently, `systemd-detect-virt --vm` returns microsoft even for the nspawn boot. Not sure I like that behavior…

**Review by bluca — APPROVED** [review]

**Comment by medhefgo** [comment]
Looks like it works. qemu boots twice, kernel reports secure boot and so does the userspace test.

**Comment by bluca** [comment]
technically adds a new parameter value, but given it's useful to increase test coverage, let's not wait

**Pull request merged**